packages/react-components/src/components/RTE/RichTextEditor.tsx

```diff
@@ -55,30 +62,72 @@ export const RichTextEditor = React.forwardRef<RichTextEditorComponentRef, RichT
     }
   }, [content]);
 
-  const editorCreator = useMemo(() => {
-    return (div: HTMLDivElement) => {
-      const contentEdit = new ContentEdit();
-      const placeholderPlugin = new Watermark(placeholderText || '');
-      const updateContentPlugin = createUpdateContentPlugin(
-        UpdateMode.OnContentChangedEvent | UpdateMode.OnUserInput,
-        (content: string) => {
-          onChange && onChange(content);
-        }
-      );
+  const ribbonPlugin = React.useMemo(() => {
+    return createRibbonPlugin();
+  }, []);
 
-      const options: EditorOptions = {
-        plugins: [placeholderPlugin, contentEdit, updateContentPlugin],
-        imageSelectionBorderColor: 'blue'
-      };
+  const editorCreator = useCallback((div: HTMLDivElement, options: EditorOptions) => {
+    editor.current = new Editor(div, options);
+    // Remove the background color of the editor
+    div.style.backgroundColor = 'transparent';
+    return editor.current;
+  }, []);
+
+  const plugins = useMemo(() => {
+    const contentEdit = new ContentEdit();
+    const placeholderPlugin = new Watermark(placeholderText || '');
+    const updateContentPlugin = createUpdateContentPlugin(
+      UpdateMode.OnContentChangedEvent | UpdateMode.OnUserInput,
+      (content: string) => {
+        onChange && onChange(content);
+      }
+    );
+    return [contentEdit, placeholderPlugin, updateContentPlugin, ribbonPlugin];
+  }, [onChange, placeholderText, ribbonPlugin]);
+
+  const ribbon = useMemo(() => {
+    const buttons = ribbonButtons(theme);
+
+    return (
+      //TODO: Add localization for watermark plugin https://github.com/microsoft/roosterjs/issues/2430
+      <Ribbon
+        styles={ribbonStyle()}
+        buttons={buttons}
+        plugin={ribbonPlugin}
+        overflowButtonProps={{
+          styles: ribbonButtonStyle(theme),
+          menuProps: {
+            items: [], // CommandBar will determine items rendered in overflow
+            isBeakVisible: false
+          }
+        }}
+        strings={ribbonButtonsStrings(strings)}
+      />
+    );
+  }, [strings, ribbonPlugin, theme]);
 
-      editor.current = new Editor(div, options);
-      return editor.current;
+  const defaultFormat: DefaultFormat = useMemo(() => {
+    return {
+      textColors: {
+        lightModeColor: lightTheme.palette?.neutralPrimary ?? 'black',
```

Comment:
Is it normal for us to have to have a fall back here? 

I noticed in EndCallButton we don't do that? Or is that maybe an issue?

Reply:
Got an idea hot to update to remove the fallback, will update the code